pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -73,6 +225,11 @@ let
           fi
         '';
       };
+
+      # tlshell is a GUI for tlmgr, which cannot be used in the Nix store
```

Comment:
Actually we have `tlmgr`, since I have tacked `texlive.infra` a few hours ago. It may be non functional due to it expecting a texlive.tlpdb somewhere, but we can fix it as we go along. tlmgr for local use (say tlcontrib) would be cool.

Reply:
Ah, and to make `tlshell` work, we must extract its `tlpkg/` folder too. Incidentally, the `tlpkg/` folderscontain plenty of `postaction`'s which maybe we are supposed to run anyway. So I have pulled out `texlive.infra` until we have a generic approach to the `tlpkg/` folder (probably a `tlType == "tlpkg"` container, to avoid polluting `$TEXMFDIST`).

Reply:
Now `tlmgr` kind of works. We are missing `config.guess` from `bin.core`, and `texlive.tlpdb` (which is massive and hard to generate properly, so we should investigate alternatives there).

Reply:
The `config.guess` file is also available in the tlpkg/installer subdirectory of texlive.infra, which is currently explicitly excluded.

We don't have to use `config.guess` from `bin.core`, it is already included in `texlive.infra` (in the `tlpkg/installer` subdirectory, which is currently excluded). With this directory  and an empty texlive.tlpdb (`postBuild = "touch $out/texlive.tlpdb"` for tlpkg in combine), `tlmgr init-usertree` as well as `tlmgr --usermode` appear to work fine.

I'm not sure we should bother creating an accurate `texlive.tlpdb` file, as it's close to useless anyway -- I can't think of any other uses than „correctly listing the installed software“.  If we want to do it anyway, just extracting the tlpobj files from the tarballs into the individual packages and concatenating these when combining is probably the only  reasonable way to obtain an accurate tlpdb. The 10MB additional space use for texlive-full should be negligible.

Reply:
Marking as resolved, but only because it can be fixed later.